celery/app/base.py

```diff
@@ -452,6 +452,7 @@ def _task_from_fun(self, fun, name=None, base=None, bind=False, **options):
                 '_decorated': True,
                 '__doc__': fun.__doc__,
                 '__module__': fun.__module__,
+                '__annotations__': fun.__annotations__,
```

Comment:
Not all Python 2.7s will have this attribute on function objects. It would be nice to have a separate unit test to ensure this all behaves as expected for recent Python 2.7s though.
```suggestion
                '__annotations__': getattr(fun, "__annotations__", None),
```